app/controllers/users_controller.rb

```diff
@@ -1,4 +1,132 @@
+require 'will_paginate/array'
 class UsersController < ApplicationController
+
+  before_action :require_login, only: [:twitter_search, :twitter_search_user, :vimeo_search, :vimeo_search_user, :instagram_search, :instagram_search_user, :twitter_subscribe, :vimeo_subscribe, :instagram_subscribe]
+
+  def twitter
+    Seemore::Application.config.twitter
+  end
+  #
+  # def vimeo
+  #   Seemore::Application.config.vimeo
+  # end
+
   def show
+    # Updates existing subscriptions with new content
+    Subscription.update_stories
+
+    @stories = []
+    if !@current_user
+      @stories = UsersHelper.default_content
+    else
+      @stories = UsersHelper.user_content(@current_user)
+    end
+    @stories = @stories.sort_by! { |story| story[:post_time] }.reverse!
+    @stories = @stories.paginate(:page => params[:page], :per_page => 15)
```

Comment:
It looks like only the final assignment of @stories needs to be an instance variable. The others are used/mutated within the method and can be local variables.